schematic/schemas/json_schema_validation_rule_functions.py

```diff
@@ -0,0 +1,367 @@
+        incompatible_rules: Other validation rules this rule can not be paired with
+        parameters: Parameters for the validation rule that need to be collected for the JSON Schema
+    """
+
+    name: ValidationRuleName
+    js_type: Optional[JSONSchemaType]
+    incompatible_rules: list[ValidationRuleName]
+    parameters: Optional[list[str]] = None
+
+
+# A dictionary of current Schematic validation rules
+#   where the keys are name fo the rule in Schematic
+#   and the values are ValidationRule objects
+_VALIDATION_RULES = {
+    "list": ValidationRule(
+        name=ValidationRuleName.LIST,
+        js_type=None,
+        incompatible_rules=[],
+    ),
+    "date": ValidationRule(
+        name=ValidationRuleName.DATE,
+        js_type=JSONSchemaType.STRING,
+        incompatible_rules=[
+            ValidationRuleName.URL,
+            ValidationRuleName.INT,
+            ValidationRuleName.FLOAT,
+            ValidationRuleName.BOOL,
+            ValidationRuleName.NUM,
+        ],
+    ),
+    "url": ValidationRule(
+        name=ValidationRuleName.URL,
+        js_type=JSONSchemaType.STRING,
+        incompatible_rules=[
+            ValidationRuleName.DATE,
+            ValidationRuleName.INT,
+            ValidationRuleName.FLOAT,
+            ValidationRuleName.BOOL,
+            ValidationRuleName.NUM,
+        ],
+    ),
+    "regex": ValidationRule(
+        name=ValidationRuleName.REGEX,
+        js_type=JSONSchemaType.STRING,
+        incompatible_rules=[
+            ValidationRuleName.INT,
+            ValidationRuleName.FLOAT,
+            ValidationRuleName.BOOL,
+            ValidationRuleName.NUM,
+        ],
+        parameters=["module", "pattern"],
+    ),
+    "inRange": ValidationRule(
+        name=ValidationRuleName.IN_RANGE,
+        js_type=JSONSchemaType.NUMBER,
+        incompatible_rules=[
+            ValidationRuleName.URL,
+            ValidationRuleName.DATE,
+            ValidationRuleName.REGEX,
+            ValidationRuleName.STR,
+            ValidationRuleName.BOOL,
+        ],
+        parameters=["minimum", "maximum"],
+    ),
+    "str": ValidationRule(
+        name=ValidationRuleName.STR,
+        js_type=JSONSchemaType.STRING,
+        incompatible_rules=[
+            ValidationRuleName.IN_RANGE,
+            ValidationRuleName.INT,
+            ValidationRuleName.FLOAT,
+            ValidationRuleName.NUM,
+            ValidationRuleName.BOOL,
+        ],
+    ),
+    "float": ValidationRule(
+        name=ValidationRuleName.FLOAT,
+        js_type=JSONSchemaType.NUMBER,
+        incompatible_rules=[
+            ValidationRuleName.URL,
+            ValidationRuleName.DATE,
+            ValidationRuleName.REGEX,
+            ValidationRuleName.STR,
+            ValidationRuleName.BOOL,
+        ],
+    ),
+    "int": ValidationRule(
+        name=ValidationRuleName.INT,
+        js_type=JSONSchemaType.INTEGER,
+        incompatible_rules=[
+            ValidationRuleName.URL,
```

Comment:
Also missing InRange?

Reply:
inRange is definitely compatible with integers